src/Device/Driver/FANET/Device.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2016 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#ifndef XCSOAR_FANET_DEVICE_HPP
+#define XCSOAR_FANET_DEVICE_HPP
+
+#include "Device/Driver.hpp"
+#include "FANET/List.hpp"
+#include "FANET/FanetStation.hpp"
+
+struct NMEAInfo;
+
+class FanetDevice final : public AbstractDevice {
+public:
+  // Virtual method from class Device
+  bool ParseNMEA(const char *line, NMEAInfo &info) override;
+
+  static bool ParseRxPacket(char *cmd_str, StationsList &stations_list, double clock);
+
+  static bool AddStationToList(StationsList &stations_list, double clock,
+                               const FanetStation &station);
+
+  static char *RemoveExtraChars(char *cmd_str);
+  static bool IntegrityCheck(char *cmd_str);
```

Comment:
What do these methods do, and why are they public?

Reply:
Added comments and made them private